govuk-one-login/authentication-frontend
# Aut 1541/playback partially obscured email

## What

On uplift journey, we'd like to display a partially obscured email to users so they know which account they're signed in with.

This functionality has been feature flagged, to allow us to release to various environments separately. (Currently on switched on in sandpit and authdevs).

I have chosen not to introduce a dependency for the email obscuring, as the obscuring is simple and I'd rather not introduce a separate dependency just for this.

If there are problems obscuring the email (it's not present, or it's invalid), we log an error and do not display anything to the user. This should not happen in practice, and should only result from a programming error, hence this error handling strategy.

Do not merge yet - needs review by UCD and security.

## How to review

For example:

1. Code Review
1. Run locally with the env variable "SHOW_REDACTED_EMAIL_ON_UPLIFT_ENABLED" set to "1", and run through an uplift journey. I actually did this by hardcoding the value of "req.session.user.upliftJourney" in the relevant controllers and running through a normal sign in journey, since I struggled to do normal uplift journey locally.
1. See that you can see an obscured email on this screen, according to the rules in the ticket.
1. Repeat for the other auth type (SMS or auth app)
1. Set the feature flag to off, and observe that you don't see this screen.


## Checklist

<!-- UCD review
